store/src/java/com/zimbra/cs/service/mail/SendMsg.java

```diff
@@ -136,6 +141,23 @@ public Element handle(Element request, Map<String, Object> context, QName respQn
                ItemId iidDraft = draftId == null ? null : new ItemId(draftId, zsc);
                boolean sendFromDraft = msgElem.getAttributeBool(MailConstants.A_SEND_FROM_DRAFT, false);
 
+               /**
+                * If the logged in user has any Persona(identity) added to his/her account 
+                * other than the DEFAULT one which is of his/her own and that identity has
+                * sendAs or sendOnBehalf rights, get the target mailbox from the identityId.
+                */
+               if (identityId != null) {
+                   Identity identity = Provisioning.getInstance().get(octxt.getAuthenticatedUser(), Key.IdentityBy.id, identityId);
+                   if (identity != null &&
+                           !identity.getAttr(Provisioning.A_zimbraPrefIdentityName).equals(ProvisioningConstants.DEFAULT_IDENTITY_NAME) &&
+                           !StringUtil.isNullOrEmpty(identity.getAttr(Provisioning.A_zimbraPrefFromAddressType)) &&
+                           (identity.getAttr(Provisioning.A_zimbraPrefFromAddressType).equals(RightConsts.RT_sendAs) ||
+                                   identity.getAttr(Provisioning.A_zimbraPrefFromAddressType).equals(RightConsts.RT_sendOnBehalfOf))) {
+                       Account targetAccount = Provisioning.getInstance().get(AccountBy.name, identity.getAttr(Provisioning.A_zimbraPrefFromAddress));
+                       mbox = MailboxManager.getInstance().getMailboxByAccount(targetAccount);
```

Comment:
Wouldn't this cause issues down below (fetching draft etc) if the account isn't local to this mailbox?

Reply:
resolved. instead of doing the same modified in `soapengine.java` to proxy the request automatically.

Reply:
Good call.